main.go

```diff
@@ -0,0 +1,146 @@
+package main
+
+import (
+	"mime"
+	"net/http"
+	"net/url"
+	"os"
+	"path/filepath"
+	"strconv"
+	"strings"
+
+	"github.com/ONSdigital/florence/assets"
+	"github.com/ONSdigital/go-ns/handlers/reverseProxy"
+	"github.com/ONSdigital/go-ns/log"
+	"github.com/ONSdigital/go-ns/server"
+	"github.com/gorilla/pat"
+)
+
+var bindAddr = ":8081"
+var babbageURL = "http://localhost:8080"
+var zebedeeURL = "http://localhost:8082"
+var enableNewApp = false
+
+var getAsset = assets.Asset
+
+func main() {
+
+	if v := os.Getenv("BIND_ADDR"); len(v) > 0 {
+		bindAddr = v
+	}
+	if v := os.Getenv("BABBAGE_URL"); len(v) > 0 {
+		babbageURL = v
+	}
+	if v := os.Getenv("ZEBEDEE_URL"); len(v) > 0 {
+		zebedeeURL = v
+	}
+	if v := os.Getenv("ENABLE_NEW_APP"); len(v) > 0 {
+		enableNewApp, _ = strconv.ParseBool(v)
+	}
+
+	log.Namespace = "florence"
+
+	/*
+		NOTE:
+		If there's any issues with this Florence server proxying redirects from either Babbage or Zebedee then the code in the previous Java Florence server might give some clues for a solution: https://github.com/ONSdigital/florence/blob/b13df0708b30493b98e9ce239103c59d7f409f98/src/main/java/com/github/onsdigital/florence/filter/Proxy.java#L125-L135
+
+		The code has purposefully not been included in this Go replacement because we can't see what issue it's fixing and whether it's necessary.
+	*/
+
+	babbageURL, err := url.Parse(babbageURL)
+	if err != nil {
+		log.Error(err, nil)
+		os.Exit(1)
+	}
+
+	babbageProxy := reverseProxy.Create(babbageURL, nil)
+
+	zebedeeURL, err := url.Parse(zebedeeURL)
+	if err != nil {
+		log.Error(err, nil)
+		os.Exit(1)
+	}
+
+	zebedeeProxy := reverseProxy.Create(zebedeeURL, zebedeeDirector)
+
+	router := pat.New()
+
+	newAppHandler := refactoredIndexFile
+
+	if !enableNewApp {
+		newAppHandler = legacyIndexFile
+	}
+
+	router.Handle("/zebedee/{uri:.*}", zebedeeProxy)
+	router.HandleFunc("/florence/dist/{uri:.*}", staticFiles)
+	router.HandleFunc("/florence", newAppHandler)
+	router.HandleFunc("/florence/index.html", legacyIndexFile)
+	router.HandleFunc("/florence{uri:|/.*}", newAppHandler)
+	router.Handle("/{uri:.*}", babbageProxy)
+
+	log.Debug("Starting server", log.Data{
+		"bind_addr":      bindAddr,
+		"babbage_url":    babbageURL,
+		"zebedee_url":    zebedeeURL,
+		"enable_new_app": enableNewApp,
+	})
+
+	s := server.New(bindAddr, router)
+
+	if err := s.ListenAndServe(); err != nil {
+		log.Error(err, nil)
+		os.Exit(2)
+	}
+}
+
+func staticFiles(w http.ResponseWriter, req *http.Request) {
+	path := req.URL.Query().Get(":uri")
+
+	b, err := getAsset("../dist/" + path)
+	if err != nil {
+		log.Error(err, nil)
+		w.WriteHeader(404)
+		return
+	}
+
+	w.Header().Set(`Content-Type`, mime.TypeByExtension(filepath.Ext(path)))
```

Comment:
worth looking at `http.DetectContentType`, think we've used that in other places